### PR TITLE
Add JUnit "error" test cases

### PR DESCRIPTION
**Pull request by qinqon**
testgrid mark test as passed if they are of type "error" [1]. This
change add the type "error" and store it in the Result as Errored field.

[1] https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd#L75

Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

File: metadata/junit/junit.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ type Result struct {
 	Failure    *string     `xml:"failure,omitempty"`
 	Output     *string     `xml:"system-out,omitempty"`
 	Error      *string     `xml:"system-err,omitempty"`
+	Errored    *string     `xml:"error,omitempty"`
```
Comment (michelle192837) [inline metadata/junit/junit.go:105]: I think you want to process this in the Result.Truncate() function too.

(Also wondering how easy it is to rename 'Output' and 'Error' to append 'Sys' or 'System', but think that can be a separate PR.

Comment (qinqon) [reply metadata/junit/junit.go:105]: I will add the truncate, about the renaming, it will break compatibility, but If that's not a problem I can try to rename it.

Comment (michelle192837) [reply metadata/junit/junit.go:105]: I think leaving it to a separate PR (or filing an issue about it) is fine.

Also forgot, but you'll need to update Message() as well.

Comment (qinqon) [reply metadata/junit/junit.go:105]: Done

File: metadata/junit/junit.go [hunk 2]
```diff
@@ -172,7 +175,7 @@ func truncatePointer(str *string, max int) {
 
 // Truncate ensures that strings do not exceed the specified length.
 func (r Result) Truncate(max int) {
-	for _, s := range []*string{r.Failure, r.Skipped, r.Error, r.Output} {
+	for _, s := range []*string{r.Error, r.Failure, r.Skipped, r.Output} {
```
Comment (michelle192837) [inline metadata/junit/junit.go:178]: Think this still needs to account for r.Errored too?

File: metadata/junit/junit.go [hunk 3]
```diff
@@ -130,10 +131,12 @@ func (r *Result) SetProperty(name, value string) {
 
 // Message extracts the message for the junit test case.
 //
-// Will use the first non-empty <failure/>, <skipped/>, <system-err/>, <system-out/> value.
+// Will use the first non-empty <error/>, <failure/>, <skipped/>, <system-err/>, <system-out/> value.
 func (r Result) Message(max int) string {
 	var msg string
 	switch {
+	case r.Errored != nil && *r.Errored != "":
```
Comment (michelle192837) [inline metadata/junit/junit.go:138]: Add a corresponding test case to TestMessage().

File: metadata/junit/junit.go (hunk 2 above)
Comment (qinqon) [reply metadata/junit/junit.go]: Made a mistake, fixed.

File: metadata/junit/junit.go (hunk 3 above)
Comment (qinqon) [reply metadata/junit/junit.go:138]: Done

**Review by michelle192837 — APPROVED** [review]
Thanks! ^^

**Comment by fejta** [comment]
These results will still show up as passing as testgrid until you update https://github.com/GoogleCloudPlatform/testgrid/blob/5714d1590e6c0968c1f4ab5d78a7ee6294625d9c/pkg/updater/gcs.go#L251-L255
